pkg/types/validation/installconfig.go

```diff
@@ -77,8 +77,13 @@ func ValidateInstallConfig(c *types.InstallConfig, usingAgentMethod bool) field.
 	if c.FIPS {
 		allErrs = append(allErrs, validateFIPSconfig(c)...)
 	} else if c.SSHKey != "" {
-		if err := validate.SSHPublicKey(c.SSHKey); err != nil {
-			allErrs = append(allErrs, field.Invalid(field.NewPath("sshKey"), c.SSHKey, err.Error()))
+		sshKeys := strings.Split(c.SSHKey, "\n")
+		for _, sshKey := range sshKeys {
+			if sshKey != "" {
```

Comment:
There is a case where it might look confusing. Here, there is an "empty space" on the second line.
```yaml
sshKey: |
  ssh-ed25519 AAAA-pretend-this-is-a-valid-public-key-Ta thvo@example.com
   <- there is an empty space here
  ssh-ed25519 AAAA-pretend-this-is-a-valid-public-key-Ta thvo@example.com
```
The installer fails with:
```
ERROR failed to fetch Master Machines: failed to load asset "Install Config": failed to create install config: invalid "install-config.yaml" file: sshKey: Invalid value: " ": ssh: no key found 
```

To users, it doesn't look anything wrong (i.e. to catch a single empty space haha). How about that we make sure to skip empty spaces (not just a single space)? Or perhaps pre-format the `SSHKey` field?

Reply:
I would argue that you have a badly formatted install-config file then. That error message looks correct to me, since instead of a key, there is a space ;-)

It's my opinion that if you are hand adding keys to your install-config, tread carefully ;-) 

Thoughts?

Reply:
> How about that we make sure to skip empty spaces (not just a single space)? 

In general, it's better to throw an error and let users fix input rather than try to fix it for them.

> Or perhaps pre-format the SSHKey field?

Can you expand on this?

Reply:
> Can you expand on this?

Oh I meant to suggest formatting the provided `sshKey` content early such that:

- There are no trailing and leading spaces.
- There are no empty lines (i.e. with spaces only and no other contents).
- For each line, there are no trailing and leading spaces.

This is basically what we kinda expect users to use the field, right? So, subsequent access can just split by newline to get keys.

But I don't want to complicate things here haha (and Patrick said to prefer not fixing the input).

Reply:
What would be the bug here?

Reply:
If something is badly formatted in the install-config, that is user error, not a bug.